### PR TITLE
Add season selection to menu

### PR DESCRIPTION
**Pull request by ArBridgeman**

### CODE REVIEW

**Comment by ArBridgeman** [comment]
- [ ] Add tests for new selection & update other code
- [ ] Consider if want todoist reminder or want to hardcode date ranges here

File: codecov.yml [hunk 1]
```diff
@@ -8,3 +8,23 @@ coverage:
       default:
         target: 80%
         threshold: 1%  # the leniency in hitting the target
+
+comment:
```
Comment (ArBridgeman) [inline codecov.yml:12]: Added this, but it seems a bit messed up. I don't think the coverage is universally 100% in those areas

Screenshot from 2024-10-01 11-27-01

File: sous-chef/tests/e2e_tests/test_e2e.py [hunk 2]
```diff
@@ -94,8 +94,8 @@ def test_e2e(
             return_value=frozen_due_datetime_formatter.anchor_datetime,
         ):
             with patch.object(RecipeBook, "__post_init__", return_value=None):
-
-                assert_equal_dataframe(self._run_menu(), get_final_menu())
+                final_menu = self._run_menu()
```
Comment (ArBridgeman) [inline sous-chef/tests/e2e_tests/test_e2e.py:97]: - [ ] Fix e2e tests -> not fully working atm

File: sous-chef/sous_chef/menu/create_menu/_from_fixed_template.py [hunk 3]
```diff
@@ -231,14 +237,21 @@ def _get_default_prep_datetime(self, row: pd.Series):
...
-        menu_number = self.config.menu_number
         self._check_fixed_menu_number(basic_number)
+        menu_number = self.config.menu_number
         self._check_fixed_menu_number(menu_number)
-
         fixed_menu = self.all_menus_df[
             self.all_menus_df.menu.isin([basic_number, menu_number])
         ].copy()
 
+        basic_season = self.config.basic_season.casefold()
```
Comment (ArBridgeman) [inline sous-chef/sous_chef/menu/create_menu/_from_fixed_template.py:247]: add tests for filters applied here for menu & season selection

File: sous-chef/tests/e2e_tests/test_e2e.py (hunk 2 above)
Comment (ArBridgeman) [reply sous-chef/tests/e2e_tests/test_e2e.py]: @AkagiShigeru can you confirm that these now succeed locally for you?
Should we invest time to make sure these run with the CI, so we don't run into problems again?
Tbf, we haven't had as many issues in the past year with upgrades, likely due to the other coverage, but they'd still be nice to have to ensure the inputs-outputs work together as expected.

File: codecov.yml (hunk 1 above)
Comment (ArBridgeman) [reply codecov.yml:12]: Nvm. this seems to be patch-related coverage. It now looks different/ok.

Comment (ArBridgeman) [reply codecov.yml:12]: If it works, we could include running the tests for the other 2 test/coverage workflows.

File: sous-chef/tests/e2e_tests/test_e2e.py (hunk 2 above)
Comment (ArBridgeman) [reply sous-chef/tests/e2e_tests/test_e2e.py]: Added them to hopefully only on merge to main since they take awhile.

**Comment by ArBridgeman** [comment]
Also change extra menu prep tasks to only be created if we make [whole] or schedule separately.

File: sous-chef/tests/e2e_tests/test_e2e.py [hunk 4]
```diff
@@ -93,28 +93,32 @@ def test_e2e(
...
+
+        menu_history._load_history()
+        assert_equal_dataframe(menu_history.dataframe, get_menu_history())
+
+        tasks_menu = self._convert_task_list_to_df(
+            todoist_helper=todoist_helper
+        )
+        assert_equal_dataframe(tasks_menu, get_tasks_menu())
+
+        #     with patch.object(RecipeBook, "__post_init__", return_value=None):
```
Comment (ArBridgeman) [inline sous-chef/tests/e2e_tests/test_e2e.py:108]:
```suggestion
```

File: sous-chef/sous_chef/abstract/extended_enum.py [hunk 5]
```diff
@@ -20,6 +20,10 @@ class ExtendedEnum(Enum):
     def name_list(cls, string_method: str = "casefold"):
         return list(map(lambda c: getattr(c.name, string_method)(), cls))
 
+    @classmethod
```
Comment (ArBridgeman) [inline sous-chef/sous_chef/abstract/extended_enum.py:23]: could add test for, not critical atm

Comment (AkagiShigeru) [reply sous-chef/sous_chef/abstract/extended_enum.py:23]: added

**Review by AkagiShigeru — APPROVED** [review]
Looking good to me :)

**Pull request merged**